src/query/ast/src/ast/expr.rs

```diff
@@ -227,6 +227,11 @@ pub enum Expr {
         span: Span,
         name: String,
     },
+    /// `?` or `$` Prepared statement arg placeholder
+    Placeholder {
+        span: Span,
+        name: String,
```

Comment:
Consider adding an offset field, which would make the parameter binding easier to implement

FYI:
https://github.com/matrixorigin/matrixone/blob/bd8054941cfdab4cd94d17cd4260d11833a102b2/pkg/sql/parsers/dialect/mysql/mysql_lexer.go#L83-L86

https://github.com/matrixorigin/matrixone/pull/3957/files#diff-6f8d2e3805b41b9704c5d638c08078e8a7e5f6812ffa344ad607746319026e1fR856